core/src/main/java/org/infinispan/executors/LazyInitializingBlockingTaskAwareExecutorService.java

```diff
@@ -33,8 +35,50 @@
    private static final BlockingTaskAwareExecutorService STOPPED;
 
    static {
-      STOPPED = new BlockingTaskAwareExecutorServiceImpl("", new WithinThreadExecutor(), null);
-      STOPPED.shutdown();
+      STOPPED = new EmptyBlockingTaskAwareExecutorService();
+   }
+
+   static final class EmptyBlockingTaskAwareExecutorService extends AbstractExecutorService implements BlockingTaskAwareExecutorService {
```

Comment:
Could you add a comment/javadoc to explain why the new class is needed?

Reply:
I added it to prevent the starting of a thread during static initialization time (BlockingTaskAwareExecutorServiceImpl does this in the constructor).

I can add the comment on the next PR of changes.